packages/instant/src/index.umd.ts

```diff
@@ -156,14 +156,17 @@ export const ERC721_PROXY_ID = AssetProxyId.ERC721;
 
 export const ERC20_PROXY_ID = AssetProxyId.ERC20;
 
-export const assetDataForERC20TokenAddress = (tokenAddress: string): string => {
+export const assetDataForERC20TokenAddressAsync = async (tokenAddress: string): Promise<string> => {
     assert.isETHAddressHex('tokenAddress', tokenAddress);
-    return assetDataUtils.encodeERC20AssetData(tokenAddress);
+    return devUtilsContract.encodeERC20AssetData(tokenAddress).callAsync();
 };
 
-export const assetDataForERC721TokenAddress = (tokenAddress: string, tokenId: string | number): string => {
+export const assetDataForERC721TokenAddressAsync = async (
+    tokenAddress: string,
+    tokenId: string | number,
+): Promise<string> => {
     assert.isETHAddressHex('tokenAddress', tokenAddress);
-    return assetDataUtils.encodeERC721AssetData(tokenAddress, new BigNumber(tokenId));
+    return devUtilsContract.encodeERC721AssetData(tokenAddress, new BigNumber(tokenId)).callAsync();
```

Comment:
This is a significant breaking change. Obviously I'm lacking context on this decision but it seems odd that you can no longer get assetDatas in a synchronous way.

Reply:
That is a good point @fragosti. @xianny we could change the method signature in the wrapper to be synchronous if the method is pure, no?

Reply:
Not easily, we currently use `putContractCode` and `runCall` from `ethereumjs-vm` and they are async. 

@jalextowle showed me a synchronous work-around using `IAssetDataContract` though:

```
const assetDataEncoder = new IAssetDataContract(NULL_ADDRESS, fakeProvider);
const encoded = assetDataEncoder.ERC20Token(tokenAddress).getABIEncodedTransactionData()
```